memory/src/main/scala/filodb.memory/format/vectors/DoubleVector.scala

```diff
@@ -144,15 +152,30 @@ trait DoubleVectorDataReader extends VectorDataReader {
  */
 object DoubleVectorDataReader64 extends DoubleVectorDataReader {
   import PrimitiveVector.OffsetData
-  final def apply(vector: BinaryVectorPtr, n: Int): Double = UnsafeUtils.getDouble(vector + OffsetData + n * 8)
-  def iterate(vector: BinaryVectorPtr, startElement: Int = 0): DoubleIterator = new DoubleIterator {
-    private final var addr = vector + OffsetData + startElement * 8
+
+  class Double64Iterator(var addr: BinaryRegion.NativePointer) extends DoubleIterator {
     final def next: Double = {
       val data = UnsafeUtils.getDouble(addr)
       addr += 8
       data
     }
   }
+
+  final def apply(vector: BinaryVectorPtr, n: Int): Double = UnsafeUtils.getDouble(vector + OffsetData + n * 8)
+  def iterate(vector: BinaryVectorPtr, startElement: Int = 0): DoubleIterator =
+    new Double64Iterator(vector + OffsetData + startElement * 8)
+
+  // end is inclusive
+  final def sum(vector: BinaryVectorPtr, start: Int, end: Int): Double = {
```

Comment:
One important concern. How is NaN handled ?
NaNs can occur for end-of-time-series markers. 
Can you please see how the performance is impacted because of any additional NaN check ?

Reply:
Sure thing.